src/js/pages/main.js

```diff
@@ -0,0 +1,134 @@
+import { getProducts } from '../mocks/products.js';
+import { setListener } from '../utils/dom.js';
+import { formatToBRL } from '../utils/currency.js';
+
+class Main {
+  constructor() {
+    this.products = [];
+    this.filters = {
+      category: 'all',
+      search: '',
+      price: 'asc',
+    };
+    this.renderElement = this.getRenderElement();
+
+    this.init();
+  }
+
+  init() {
+    this.setCategoryFieldListener();
+    this.setSearchFieldListener();
+    this.setPriceFieldListener();
+
+    this.products = getProducts();
+
+    this.applyFilters();
+  }
+
+  getRenderElement() {
+    return document.querySelector('#product-list');
+  }
+
+  setCategoryFieldListener() {
+    setListener(
+      '#category-field',
+      'change',
+      this.handleCategoryFilter.bind(this)
+    );
+  }
+
+  handleCategoryFilter(event) {
+    this.filters.category = event.target.value;
+
+    this.applyFilters();
+  }
+
+  filterByCategory(product) {
+    if (this.filters.category === 'all') {
+      return true;
+    }
+
+    return product.category === this.filters.category;
+  }
+
+  setSearchFieldListener() {
+    setListener(
+      '#search-field',
+      'input',
+      this.handleSearchFilter.bind(this)
+      );
+  }
+
+  handleSearchFilter(event) {
+    const search = event.target.value.trim();
+    const searchLength = search.length;
+
+    if (searchLength === 0 || searchLength > 1) {
```

Comment:
Dúvida: tem necessidade desse `if`? Pelo que eu entendi devemos executar a busca qualquer jeito. Se o `searchLength` for `0` (sem termo de busca) e `searchLength` for maior que `1` (com termo de busca).

Reply:
Entendi!

Então isso já resolveria:
```js
  handleSearchFilter(event) {
    const search = event.target.value.trim();
    const searchLength = search.length;
    
    if (searchLength === 1) {
       return;
    }

    this.filters.search = search;

    this.applyFilters();
  }
```

Mas é melhor deixar dessa forma, porque hoje a gente dispara a busca no 2 caractér. Se amanhã executarmos no 3, é mais fácil de alterar com o `if` atual.

Muito bom!